src/Generating/Trees.h

```diff
@@ -61,8 +61,8 @@ void GetSmallAppleTreeImage(int a_BlockX, int a_BlockY, int a_BlockZ, cNoise & a
 /** Generates an image of a large (branching) apple tree */
 void GetLargeAppleTreeImage(int a_BlockX, int a_BlockY, int a_BlockZ, cNoise & a_Noise, int a_Seq, sSetBlockVector & a_LogBlocks, sSetBlockVector & a_OtherBlocks);
 
-/** Generates a branch for a large apple tree */
-void GetLargeAppleTreeBranch(int a_BlockX, int a_BlockY, int a_BlockZ, int a_BranchLength, Vector3d a_StartDirection, Vector3d a_Direction, int a_TreeHeight, cNoise & a_Noise, sSetBlockVector & a_LogBlocks);
+/** Generates a branch. Returns the position where the branch ends. */
+Vector3d GetTreeBranch(BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, int a_BlockX, int a_BlockY, int a_BlockZ, int a_BranchLength, Vector3d a_StartDirection, Vector3d a_Direction, int a_TreeHeight, sSetBlockVector & a_LogBlocks);
```

Comment:
This is getting a ridiculous amount of parameters and many of them make no sense at the first glance. This should be clarified in the doxy-comment.
Also, "generate" should be explained as "fills a_LogBlocks with ???", and "branch" means just the log, or the leaves, too?

Reply:
> Also, "generate" should be explained as "fills a_LogBlocks with ???"

In that case shouldn't all the other tree image methods be changes as well?

Also, how about I replace all the `a_BlockXYZ` with a `Vector3i`? That will reduce the amount of parameters and will be more in line with #3853.

Reply:
Any change in documentation that makes it clearer is good.

`Vector3i` change sounds good as well.